src/main/java/budgetflow/command/AddIncomeCommand.java

```diff
@@ -0,0 +1,124 @@
+package budgetflow.command;
+
+import budgetflow.exception.InvalidNumberFormatException;
+import budgetflow.exception.MissingDateException;
+import budgetflow.exception.MissingAmountException;
+import budgetflow.exception.MissingCategoryException;
+import budgetflow.exception.MissingIncomeException;
+import budgetflow.expense.ExpenseList;
+import budgetflow.income.Income;
+import java.util.List;
+import java.util.logging.Logger;
+import budgetflow.parser.DateValidator;
+
+//@@author thienkimtranhoang
+public class AddIncomeCommand extends Command {
+    public static final String ERROR_INVALID_DATE = "Error: Date is not a valid date";
+    
+    private static final Logger logger = Logger.getLogger(AddIncomeCommand.class.getName());
+    private static final String ADD_COMMAND_PREFIX = "add ";
+    private static final int ADD_COMMAND_PREFIX_LENGTH = ADD_COMMAND_PREFIX.length();
+
+    private static final String ERROR_MISSING_INCOME_CATEGORY = "Error: Income category is required.";
+    private static final String ERROR_MISSING_INCOME_AMOUNT = "Error: Income amount is required.";
+    private static final String ERROR_MISSING_INCOME_DATE = "Error: Income date is required.";
+    private static final String ERROR_INCORRECT_INCOME_DATE = "Error: Income date is in wrong format." +
+            "please use DD-MM-YYYY format.";
+
+
+
+    public AddIncomeCommand(String input) {
+        super(input);
+        this.commandType = CommandType.CREATE;
+    }
+
+    /**
+     * Add the user income and save it to the income lists
+     *
+     * @param incomes list of incomes
+     * @throws MissingDateException if user did not provide the date for income, or when use miss the date tag
+     * @throws InvalidNumberFormatException if user enter income value in incorrect number format
+     * @throws MissingAmountException if user did not provide any amount for income or use miss the amount tag
+     * @throws MissingCategoryException if user did not provide category or miss tag for category
+     */
+    @Override
+    public void execute(List<Income> incomes, ExpenseList expenseList) throws MissingDateException,
+            InvalidNumberFormatException, MissingAmountException, MissingCategoryException,
+            MissingIncomeException {
+        Income income = extractIncome(input);
+        incomes.add(income);
+        this.outputMessage = "Income added: " + income.getCategory() + ", Amount: $" +
+                String.format("%.2f", income.getAmount()) + ", Date: " + income.getDate();
+        logger.info("Income added successfully: " + income);
+    }
+
+
+    private Income extractIncome(String input) throws InvalidNumberFormatException,
```

Comment:
Method is >30LOC. Try to refactor parts of it to new methods to reduce the length.